runtime/src/main/java/org/corfudb/runtime/collections/TxnContext.java

```diff
@@ -0,0 +1,325 @@
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.function.BiFunction;
+import java.util.function.Predicate;
+
+/**
+ * TxnContext is the access layer for binding all the CorfuStore CRUD operations.
+ * It can help reduce the footprint of a CorfuStore transaction by only having writes in it.
+ * All mutations/writes are aggregated and applied at once a the time of commit() call
+ * where a real corfu transaction is started.
+ *
+ * Created by hisundar, @wenbinzhu, @pankti-m on 2020-09-15
+ */
+@Slf4j
+public class TxnContext {
+
+    private final ObjectsView objectsView;
+    private final TableRegistry tableRegistry;
+    private final String namespace;
+    private final IsolationLevel isolationLevel;
+    private final List<Runnable> operations;
+
+    /**
+     * Creates a new TxnContext.
+     *
+     * @param objectsView   ObjectsView from the Corfu client.
+     * @param tableRegistry Table Registry.
+     * @param namespace     Namespace boundary defined for the transaction.
+     */
+    @Nonnull
+    TxnContext(@Nonnull final ObjectsView objectsView,
+               @Nonnull final TableRegistry tableRegistry,
+               @Nonnull final String namespace,
+               @Nonnull final IsolationLevel isolationLevel) {
+            this.objectsView = objectsView;
+        this.tableRegistry = tableRegistry;
+        this.namespace = namespace;
+        this.isolationLevel = isolationLevel;
+        this.operations = new ArrayList<>();
+        if (TransactionalContext.isInTransaction()) {
+            log.error("Cannot start new transaction in this thread without ending previous one");
+            throw new TransactionAlreadyStartedException(TransactionalContext.getRootContext().toString());
+        }
+    }
+
+    public  <K extends Message, V extends Message, M extends Message>
+    Table<K, V, M> getTable(@Nonnull final String tableName) {
+        return this.tableRegistry.getTable(this.namespace, tableName);
+    }
+
+    /**
+     *************************** WRITE APIs *****************************************
+     */
+
+    /**
+     * put the value on the specified key create record if it does not exist.
+     *
+     * @param table Table object to perform the create/update on.
+     * @param key   Key of the record.
+     * @param value Value or payload of the record.
+     * @param metadata Metadata associated with the record.
+     * @param <K>   Type of Key.
+     * @param <V>   Type of Value.
+     * @param <M>   Type of Metadata.
+     * @return TxnContext instance.
+     */
+    @Nonnull
+    public <K extends Message, V extends Message, M extends Message>
+    TxnContext put(@Nonnull Table<K, V, M> table,
+                   @Nonnull final K key,
+                   @Nonnull final V value,
+                   @Nullable final M metadata) {
+        operations.add(() -> {
+            table.put(key, value, metadata);
+        });
+        return this;
+    }
```

Comment:
Do we need to check the namespace of the table here or maybe in shim layer? Since now user can pass any table instead of a table name.

Reply:
Good catch. Will add this check and throw an exception if caller tries to insert data of one table into a different namespace.